subxt/src/config/signed_extensions.rs

```diff
@@ -271,18 +284,20 @@ impl<T: Config> ExtrinsicParams<T> for ChargeAssetTxPayment {
 
 impl ExtrinsicParamsEncoder for ChargeAssetTxPayment {
     fn encode_extra_to(&self, v: &mut Vec<u8>) {
-        (self.tip, self.asset_id).encode_to(v);
+        self.encode_to(v);
     }
 }
 
 impl<T: Config> SignedExtension<T> for ChargeAssetTxPayment {
     const NAME: &'static str = "ChargeAssetTxPayment";
+    type Decoded = Self;
 }
 
 /// The [`ChargeTransactionPayment`] signed extension.
-#[derive(Debug)]
+#[derive(Debug, Encode, DecodeAsType)]
 pub struct ChargeTransactionPayment {
-    tip: Compact<u128>,
+    /// Tip
+    pub tip: Compact<u128>,
```

Comment:
As above I'd expose a `tip()` getter that returns `u128` to hide the `Compact` implementation detail.